test/interface_test.py

```diff
@@ -23,13 +23,21 @@
 from luigi.interface import _WorkerSchedulerFactory
 from luigi.worker import Worker
 from luigi.interface import core
+from luigi.execution_summary import LuigiStatusCode
 
 from mock import Mock, patch, MagicMock
 from helpers import LuigiTestCase, with_config
 
 luigi.notifications.DEBUG = True
 
 
+# Using this updatable dict to make inline dict updates for beautiful tests
+class UpdatableDict(dict):
```

Comment:
Interesting idea, but this is going to be confusing as you later do `.update()` and *do* get a return value (most readers aren't used to that).


What about instead of 

         fake_summary_dict.return_value = self._create_empty_summary_dict().update({
            'ever_failed': [self.task_a]
        })

You can do

         fake_summary_dict.return_value = self._create_summary_dict_with({
            'ever_failed': [self.task_a]
        })

Reply:
Makes sense! More readable! 👍